app/javascript/utilities/viewport.js

```diff
@@ -0,0 +1,24 @@
+/**
```

Comment:
Refactor (non-blocking): Not in this PR, but in a future PR, this can be replaced with the [Intersection Observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) API it's corresponding [polyfill](https://github.com/w3c/IntersectionObserver/tree/master/polyfill) for Firefox on Android.